be/src/vec/exec/format/parquet/vparquet_group_reader.cpp

```diff
@@ -751,6 +755,38 @@ Status RowGroupReader::_read_empty_batch(size_t batch_size, size_t* read_rows, b
     return Status::OK();
 }
 
+
+Status RowGroupReader::_get_current_batch_row_id(size_t read_rows, std::vector<rowid_t>& row_ids) {
+    row_ids.resize(read_rows);
+    int64_t idx = 0;
+    int64_t read_range_rows = 0;
+    for (auto& range : _read_ranges) {
+        if (idx > read_rows) {
+            break;
+        }
+        if (read_range_rows >= _total_read_rows) {
+            for (auto x = range.first_row; x < range.last_row; x ++) {
+                row_ids[idx++] = (rowid_t) (x + _current_row_group_idx.first_row);
+            }
+        }
+        read_range_rows += range.last_row - range.first_row;
+    }
+    return Status::OK();
+}
+
+Status RowGroupReader::_fill_row_id_columns(Block* block,size_t read_rows) {
+    if (_row_id_column_iterator_pair.first != nullptr) {
+        vector<rowid_t> row_ids;
```

Comment:
the function could be call multi thread? if not, `row_ids` better be a membership of `vparquet_group_reader`